### PR TITLE
[Bug] Fix root logger handler

### LINKED ISSUE #220: Handler added to root logger
This library adds a `RichHandler` to the root logger through `logging.basicConfig` every time `get_logger` is called (which seems to be every time any common class/object is initialized):
https://github.com/Clarifai/clarifai-python/blob/dbdcb493bdb9255244276086bd193a305e2cbcc0/clarifai/utils/logging.py#L76-L90

There doesn't seem to be any option to configure this, and this can cause conflicts with a user's own logging setup and/or those of other libraries. (Especially since `RichHandler` seems to add a lot of formatting that a user's own logging setup might not support)

Instead, this library should configure and add handlers to its own loggers instead of the root logger.

### PR DESCRIPTION
**Pull request by sainivedh**
## Why
 - Fixes #220 
 
## What
 - Instead of adding `RichHandler` to root logging, now it is being added to SDK specific loggers
 
## Tests
 - Added test_misc.py for the same 
 
image

### CODE REVIEW

File: clarifai/utils/logging.py [hunk 1]
```diff
@@ -73,11 +73,21 @@ def _get_library_name() -> str:
   return __name__.split(".")[0]
 
 
-def _configure_logger(logger_level: str = "ERROR") -> None:
-  logging.basicConfig(
-      level=logger_level,
-      datefmt='%Y-%m-%d %H:%M:%S',
```
Comment (isaac-chung) [inline clarifai/utils/logging.py:79]: How is the date format set now?

File: clarifai/utils/logging.py [hunk 2]
```diff
@@ -73,11 +73,21 @@ def _get_library_name() -> str:
...
-      handlers=[RichHandler(rich_tracebacks=True)])
+def _configure_logger(name: str, logger_level: str = "ERROR") -> None:
```
Comment (isaac-chung) [inline clarifai/utils/logging.py:76]:
```suggestion
def _configure_logger(name: str, logger_level: str = logging.ERROR) -> None:
```

File: clarifai/utils/logging.py [hunk 3]
```diff
@@ -73,11 +73,21 @@ def _get_library_name() -> str:
...
+    logger.removeHandler(handler)
+
+  # Add the new rich handler and formatter
+  handler = RichHandler(rich_tracebacks=True)
+  formatter = logging.Formatter('%(name)s:  %(message)s')
+  handler.setFormatter(formatter)
+  logger.addHandler(handler)
 
 
 def get_logger(logger_level: str = "ERROR", name: Optional[str] = None) -> logging.Logger:
```
Comment (isaac-chung) [inline clarifai/utils/logging.py:93]:
```suggestion
def get_logger(logger_level: str = logging.ERROR, name: Optional[str] = None) -> logging.Logger:
```

File: clarifai/utils/logging.py (hunk 1 above)
Comment (sainivedh) [reply clarifai/utils/logging.py:79]: Great catch. Added back the same format now.

**Review by isaac-chung — APPROVED** [review]

**Review by sanjaychelliah — APPROVED** [review]

**Pull request merged**